clean.sh

```diff
@@ -44,11 +46,14 @@ fi
 		chmod +w -R "$TERMUX_TOPDIR" || true
 	fi
 
-	if $TERMUX_ON_DEVICE_BUILD; then
+	if ! $TERMUX_ON_DEVICE_BUILD; then
 		# For on-device build cleanup /data shouldn't be erased.
-		rm -Rf "$TERMUX_TOPDIR"
-	else
-		find /data -mindepth 1 ! -regex '^/data/data/com.termux/cgct\(/.*\)?' -delete 2> /dev/null || true
-		rm -Rf "$TERMUX_TOPDIR"
+		if [[ ! "$TERMUX_BASE_DIR" =~ ^(/[^/]+)+$ ]]; then
+			echo "TERMUX_BASE_DIR '$TERMUX_BASE_DIR' is not an absolute path under rootfs '/'." 1>&2
+			exit 1
+		fi
+		find "${TERMUX_BASE_DIR%"${TERMUX_BASE_DIR#/*/}"}" -mindepth 1 \
+			! -regex "^${CGCT_DIR}\(/.*\)?" -delete 2> /dev/null || true
```

Comment:
What is the parameter expansion doing here, there was a reason I gave you the code which sets a local variable with a comment first. Additionally, environment variable with paths must not be used in regexes, they can contain special characters, which can make regex mean something else than what is intended, that is why my code used `-path` instead.

Reply:
You can use following

```shell
# Escape '\$[](){}|^.?+*' with backslashes
cgct_dir_escaped="$(printf "%s" "$CGCT_DIR" | sed -zE -e 's/[][\.|$(){}?+*^]/\\&/g')"
find "$rootfs_top_level_dir" -mindepth 1 -regextype posix-extended ! -regex "^$cgct_dir_escaped(/.*)?" -delete 2>/dev/null || true
```